pkg/graph/resolvers/plan_collaborator.go

```diff
@@ -161,8 +162,39 @@ func PlanCollaboratorDelete(logger *zap.Logger, id uuid.UUID, principal authenti
 	if err != nil {
 		return nil, err
 	}
-	retCollaborator, err := store.PlanCollaboratorDelete(logger, id, principal.Account().ID)
-	return retCollaborator, err
+
+	return sqlutils.WithTransaction(store, func(tx *sqlx.Tx) (*models.PlanCollaborator, error) {
+		// First, update any milestones assigned to this collaborator
+		updatedMilestones, err := storage.MTOMilestoneUpdateAssignedToNullByCollaborator(
```

Comment:
Hmm is this necessary because the assignement points to a user not a collaborator? Does a milestone have to be a assigned to a collaborator? Or could anyone be assigned to the milestone? If only a collaborator could be assigned, this could just be cascade logic.